### PR TITLE
Creating ES5 version of the TS client

### PR DESCRIPTION
**Pull request by moozzyk**
Here are the sizes:
```
Mode                LastWriteTime         Length Name
----                -------------         ------ ----
-a----        8/25/2017   4:42 PM          32339 signalr-client.js
-a----        8/25/2017   4:42 PM          14825 signalr-client.min.js
-a----        8/25/2017   4:42 PM         169458 signalr-clientES5.js
-a----        8/25/2017   4:42 PM          72055 signalr-clientES5.min.js
-a----        8/25/2017   4:42 PM         282224 signalr-msgpackprotocol.js
-a----        8/25/2017   4:42 PM         109505 signalr-msgpackprotocol.min.js
```

I also need to test ES5 versions on IE11 which is trickier than it sound since it requires transpiling the tests.

### CODE REVIEW

**Comment by moozzyk** [comment]
Needs more work - most of the tests fail in IE

**Comment by moozzyk** [comment]
:up::date:
image

File: client-ts/Microsoft.AspNetCore.SignalR.Client.TS/Transports.ts [hunk 1]
```diff
@@ -248,6 +245,11 @@ export class LongPollingTransport implements ITransport {
 
         this.pollXhr = pollXhr;
         this.pollXhr.open("GET", url, true);
+        if (transferMode === TransferMode.Binary) {
+            pollXhr.responseType = "arraybuffer";
```
Comment (BrennanConroy) [inline client-ts/Microsoft.AspNetCore.SignalR.Client.TS/Transports.ts]: Most are `this.pollXhr` this one isn't?

Comment (moozzyk) [reply client-ts/Microsoft.AspNetCore.SignalR.Client.TS/Transports.ts]: Moved from a different place where this.pollXhr was not assigned. I will fix for consistency.

File: client-ts/Microsoft.AspNetCore.SignalR.Test.Server/wwwroot/js/hubConnectionTests.js [hunk 2]
```diff
@@ -1,194 +1,149 @@
...
-                const message = "你好 SignalR！";
+                var message = "你好 SignalR！";
 
-                let callbackPromise = new Promise((resolve, reject) => {
-                    hubConnection.on("Message", msg => {
-                        expect(msg).toBe(message);
-                        resolve();
-                    });
+                hubConnection.on("Message", function (msg) {
+                    expect(msg).toBe(message);
```
Comment (BrennanConroy) [inline client-ts/Microsoft.AspNetCore.SignalR.Test.Server/wwwroot/js/hubConnectionTests.js]: So we are no longer guaranteed to be asserting this?

Comment (moozzyk) [reply client-ts/Microsoft.AspNetCore.SignalR.Test.Server/wwwroot/js/hubConnectionTests.js]: We are - the test won't complete without invoking this callback (unless start/invoke throws which will cause a test failure). If the callback is not invoked within the timeout (5 secs by default) the test will fail.

**Pull request merged**